R/create_landscape_covariates.R

```diff
@@ -0,0 +1,345 @@
+    #                     - once this is stlyed then you can adjust the grouping on the DAH to match
+    #                     - Deception = -0.2 to 0.2.
+    #                     - Date Creek = -0.3 to 0.3.
+    #                     - Peter Hope = -0.2 to 0.2
+
+    m <- c(
+      -10, (dah_threshold * -1), 1,
+      (dah_threshold * -1), dah_threshold, 2,
+      dah_threshold, 10, 3
+    )
+
+    rclmat <- matrix(m, ncol = 3, byrow = TRUE)
+
+    rc <- terra::classify(dah_r, rclmat)
+
+    rc <- rc |>
+      terra::sieve(threshold = sieve_size, directions = 8)
+
+    rc_crop <- terra::crop(rc, dtm)
+
+    terra::writeRaster(rc_crop, fs::path(output_dir, "dah_LS.tif"), overwrite = TRUE)
+  }
+
+  ## 3. Landform Class
+
+  land_class <- create_landform_classes(dtm, scale = 75, sn = 3, ln = 7, n.classes = "six")
+
+  land_class <- land_class |>
+    terra::sieve(threshold = sieve_size, directions = 8) |>
+    terra::subst(from = 0, to = NA)
+
+  land_class <- terra::crop(land_class, dtm)
+  names(land_class) <- "landclass"
+
+  terra::writeRaster(land_class, file.path(output_dir, "landform_LS.tif"), overwrite = TRUE)
+
+  # remove temp Saga files
+  # unlink(paste(output, "saga", sep = "/"), recursive = TRUE)
+  # delete the files
+
+  ## decide if want to delete these temp files.
+
+  # to_delete <- grep(".tif", list.files(file.path(output_dir), full.names = TRUE), value = T, invert = TRUE)
+  # file.remove(to_delete)
+
+  cli::cat_line()
+  cli::cli_alert_success(
+    "Layers downloaded and to written to {.path {out_dir}}"
+  )
+  invisible(out_dir)
+}
+
+
+create_landform_classes <- function(dtm,
+                                    scale = 75,
+                                    sn = 3,
+                                    ln = 7,
+                                    n.classes = "ten",
+                                    add.tpi = FALSE,
+                                    stand.tpi = FALSE) {
+  # calculate the slope from the input DTM, to be used for either the six or ten class slope position
+  slp <- terra::terrain(dtm, v = "slope", unit = "degrees", neighbors = 8)
+
+  win <- "rectangle"
+
+  if (n.classes == "six") {
+    # calculate the tpi using spatialEco::tpi function
+    tp <- spatialEco::tpi(dtm, scale = scale, win = win, normalize = TRUE)
+
+    # define the six classes on the basis of thresholds of tp and slope
+    valley <- (tp <= -1)
+    valley[stats::na.omit(valley)] <- 1
+
+    lower.slp <- (tp > -1 & tp <= -0.5)
+    lower.slp[stats::na.omit(lower.slp)] <- 2
+
+    flat.slp <- (tp > -0.5 & tp < 0.5) & (slp <= 5)
+    flat.slp[stats::na.omit(flat.slp)] <- 3
+
+    middle.slp <- (tp > -0.5 & tp < 0.5) & (slp > 5)
+    middle.slp[stats::na.omit(middle.slp)] <- 4
+
+    upper.slp <- (tp > 0.5 & tp <= 1)
+    upper.slp[stats::na.omit(upper.slp)] <- 5
+
+    ridge <- (tp > 1)
+    ridge[stats::na.omit(ridge)] <- 6
+
+    # plot(valley, main="Valley", sub="TPI <= -1", cex.main=0.9, cex.sub=0.7, legend=FALSE)
```

Comment:
At some point it would be good to move this commented code out of here and into a script in a `scratch` folder or something like that, it makes it a bit hard to navigate...